File: src/ghosts/Pinky.java
package ghosts;

import engine.Game;
import engine.Ghost;

public class Pinky extends Ghost {

    public Pinky(double x, double y, double speed) {
        super(x, y, speed, "pinky");
    }
    
    @Override
    public void setCible() {
        xCible = Game.getPlayers().get(0).getTileX();
        yCible = Game.getPlayers().get(0).getTileY();
        
        switch(Game.getPlayers().get(0).getDirection()){
            case Up:
                xCible-=4;
                yCible-=4;
                break;
            case Down:
                yCible+=4;
                break;
            case Left:
                xCible-=4;
                break;
            case Right:
                xCible+=4;
                break;
        }
    }
    
}

File: pacman/Ghosts/Pinky.cs
using System.Windows.Media;

namespace pacman
{
    public class Pinky : Ghost
    {
        public Pinky(Node nodeFrom, Direction direction, double distance = 0) : base(nodeFrom, direction, distance)
        {
            color = Brushes.Pink;
            scatterX = 2;
            scatterY = 0;
        }

        protected override void Init()
        {
            name = "ghosts/pinky";
        }

        public override void CalculateTargetChaseMode()
        {
            (targetX, targetY) = Game.entities.Find(e => e is Player).GetIntXY();

            switch (Game.entities.Find(e => e is Player).direction)
            {
                case Direction.Right:
                    targetX += 4;
                    break;
                case Direction.Left:
                    targetX -= 4;
                    break;
                case Direction.Down:
                    targetY += 4;
                    break;
                case Direction.Up:
                    targetY -= 4;
                    targetX -= 4;
                    break;
            }
        }
    }
}
